builtin/logical/pki/path_config_ca.go

```diff
@@ -13,24 +16,443 @@ func pathConfigCA(b *backend) *framework.Path {
+		}
+	}
+
+	cb, err := parsedBundle.ToCertBundle()
+	if err != nil {
+		return nil, fmt.Errorf("error converting raw cert bundle to cert bundle: %s", err)
+	}
+
+	resp = &logical.Response{
+		Data: map[string]interface{}{
+			"serial_number": cb.SerialNumber,
+			"expiration":    int64(parsedBundle.Certificate.NotAfter.Unix()),
+		},
+	}
+
+	switch format {
+	case "pem":
+		resp.Data["certificate"] = cb.Certificate
+		resp.Data["issuing_ca"] = cb.IssuingCA
+		if exported {
+			resp.Data["private_key"] = cb.PrivateKey
+			resp.Data["private_key_type"] = cb.PrivateKeyType
+		}
+	case "der":
+		resp.Data["certificate"] = base64.StdEncoding.EncodeToString(parsedBundle.CertificateBytes)
+		resp.Data["issuing_ca"] = base64.StdEncoding.EncodeToString(parsedBundle.IssuingCABytes)
+		if exported {
+			resp.Data["private_key"] = base64.StdEncoding.EncodeToString(parsedBundle.PrivateKeyBytes)
+			resp.Data["private_key_type"] = cb.PrivateKeyType
+		}
+	}
+
+	entry, err := logical.StorageEntryJSON("config/ca_bundle", cb)
+	if err != nil {
+		return nil, err
+	}
+	err = req.Storage.Put(entry)
+	if err != nil {
+		return nil, err
+	}
+
+	// For ease of later use, also store just the certificate at a known
+	// location, plus a fresh CRL
+	entry.Key = "ca"
+	entry.Value = parsedBundle.CertificateBytes
+	err = req.Storage.Put(entry)
+	if err != nil {
+		return nil, err
+	}
+
+	err = buildCRL(b, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if parsedBundle.Certificate.MaxPathLen == 0 {
+		resp.AddWarning("Max path length of the generated certificate is zero")
+	}
+
+	return resp, nil
+}
+
+func (b *backend) pathCAGenerateIntermediate(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	var err error
+
+	exported, format, role, errorResp := b.getGenerationParams(data)
+	if errorResp != nil {
+		return errorResp, nil
+	}
+
+	var resp *logical.Response
+	parsedBundle, err := generateCSR(b, role, nil, req, data)
+	if err != nil {
+		switch err.(type) {
+		case certutil.UserError:
+			return logical.ErrorResponse(err.Error()), nil
+		case certutil.InternalError:
+			return nil, err
+		}
+	}
+
+	csrb, err := parsedBundle.ToCSRBundle()
+	if err != nil {
+		return nil, fmt.Errorf("Error converting raw CSR bundle to CSR bundle: %s", err)
+	}
+
+	resp = &logical.Response{
+		Data: map[string]interface{}{},
+	}
+
+	switch format {
+	case "pem":
+		resp.Data["csr"] = csrb.CSR
+		if exported {
+			resp.Data["private_key"] = csrb.PrivateKey
+			resp.Data["private_key_type"] = csrb.PrivateKeyType
+		}
+	case "der":
+		resp.Data["csr"] = base64.StdEncoding.EncodeToString(parsedBundle.CSRBytes)
+		if exported {
+			resp.Data["private_key"] = base64.StdEncoding.EncodeToString(parsedBundle.PrivateKeyBytes)
+			resp.Data["private_key_type"] = csrb.PrivateKeyType
+		}
+	}
+
+	cb := &certutil.CertBundle{
+		PrivateKey:     csrb.PrivateKey,
+		PrivateKeyType: csrb.PrivateKeyType,
+	}
+
+	entry, err := logical.StorageEntryJSON("config/ca_bundle", cb)
```

Comment:
Yes. At the moment, it's up to the user not to overwrite previous CAs. The good news is that since this isn't likely to happen often, backups are likely to be useful in a break-glass scenario.